### ISSUE TITLE
LocalVariableName: false negative on pattern variable 

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#10972]
https://checkstyle.org/config_naming.html#LocalVariableName

````
$ cat TestClass.java
public class Test {
    public void run(String... arguments) {
        final Object o = 45;
        if (o instanceof String BAD) {
            System.out.println(BAD);
        }
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="LocalFinalVariableName"/>
<module name="LocalVariableName"/>
    </module>
</module>

$ java -jar checkstyle-9.0.1-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

I am expecting a violation on line 5 since BAD is a local variable and it doesn't start with a lower case as is default required.

**Comment by rnveach** [comment]
I guess this may require more discussion since we have PatternVariableName ( https://checkstyle.org/config_naming.html#PatternVariableName ) and we don't want to overlap functionality. Not sure if it makes sense split apart final and non-final patterns. At the least, we should update documentation to point to the other check for patterns.

**Comment by romani** [comment]
As we have special Check, others should exclude violating such cases.
I think it is correct to have no violations on code and config above
